lightning/src/chain/channelmonitor.rs

```diff
@@ -933,42 +935,55 @@ impl<ChanSigner: ChannelKeys + Writeable> ChannelMonitor<ChanSigner> {
 
 impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 	pub(crate) fn new(keys: ChanSigner, shutdown_pubkey: &PublicKey,
-			on_counterparty_tx_csv: u16, destination_script: &Script, funding_info: (OutPoint, Script),
-			counterparty_htlc_base_key: &PublicKey, counterparty_delayed_payment_base_key: &PublicKey,
-			on_holder_tx_csv: u16, funding_redeemscript: Script, channel_value_satoshis: u64,
-			commitment_transaction_number_obscure_factor: u64,
-			initial_holder_commitment_tx: HolderCommitmentTransaction) -> ChannelMonitor<ChanSigner> {
+	                  on_counterparty_tx_csv: u16, destination_script: &Script, funding_info: (OutPoint, Script),
+	                  channel_static_info: &ChannelStaticInfo,
+	                  funding_redeemscript: Script, channel_value_satoshis: u64,
+	                  commitment_transaction_number_obscure_factor: u64,
+	                  initial_holder_commitment_info: HolderCommitmentTransactionInfo) -> ChannelMonitor<ChanSigner> {
 
 		assert!(commitment_transaction_number_obscure_factor <= (1 << 48));
 		let our_channel_close_key_hash = WPubkeyHash::hash(&shutdown_pubkey.serialize());
 		let shutdown_script = Builder::new().push_opcode(opcodes::all::OP_PUSHBYTES_0).push_slice(&our_channel_close_key_hash[..]).into_script();
 		let payment_key_hash = WPubkeyHash::hash(&keys.pubkeys().payment_point.serialize());
 		let counterparty_payment_script = Builder::new().push_opcode(opcodes::all::OP_PUSHBYTES_0).push_slice(&payment_key_hash[..]).into_script();
 
-		let counterparty_tx_cache = CounterpartyCommitmentTransaction { counterparty_delayed_payment_base_key: *counterparty_delayed_payment_base_key, counterparty_htlc_base_key: *counterparty_htlc_base_key, on_counterparty_tx_csv, per_htlc: HashMap::new() };
-
-		let mut onchain_tx_handler = OnchainTxHandler::new(destination_script.clone(), keys.clone(), on_holder_tx_csv);
-
-		let holder_tx_sequence = initial_holder_commitment_tx.unsigned_tx.input[0].sequence as u64;
-		let holder_tx_locktime = initial_holder_commitment_tx.unsigned_tx.lock_time as u64;
-		let holder_commitment_tx = HolderSignedTx {
-			txid: initial_holder_commitment_tx.txid(),
-			revocation_key: initial_holder_commitment_tx.keys.revocation_key,
-			a_htlc_key: initial_holder_commitment_tx.keys.broadcaster_htlc_key,
-			b_htlc_key: initial_holder_commitment_tx.keys.countersignatory_htlc_key,
-			delayed_payment_key: initial_holder_commitment_tx.keys.broadcaster_delayed_payment_key,
-			per_commitment_point: initial_holder_commitment_tx.keys.per_commitment_point,
-			feerate_per_kw: initial_holder_commitment_tx.feerate_per_kw,
-			htlc_outputs: Vec::new(), // There are never any HTLCs in the initial commitment transactions
+		let counterparty_pubkeys = &channel_static_info.counterparty_pubkeys;
```

Comment:
This diff without this line works fine for me ?

```
diff --git a/lightning/src/chain/channelmonitor.rs b/lightning/src/chain/channelmonitor.rs
index 51528b34..a71645b9 100644
--- a/lightning/src/chain/channelmonitor.rs
+++ b/lightning/src/chain/channelmonitor.rs
@@ -947,9 +947,8 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
                let payment_key_hash = WPubkeyHash::hash(&keys.pubkeys().payment_point.serialize());
                let counterparty_payment_script = Builder::new().push_opcode(opcodes::all::OP_PUSHBYTES_0).push_slice(&payment_key_hash[..]).into_script();
 
-               let counterparty_pubkeys = &channel_static_info.counterparty_pubkeys;
-               let counterparty_delayed_payment_base_key = counterparty_pubkeys.delayed_payment_basepoint;
-               let counterparty_htlc_base_key = counterparty_pubkeys.htlc_basepoint;
+               let counterparty_delayed_payment_base_key = channel_static_info.counterparty_pubkeys.delayed_payment_basepoint;
+               let counterparty_htlc_base_key = channel_static_info.counterparty_pubkeys.htlc_basepoint;
                let counterparty_tx_cache = CounterpartyCommitmentTransaction { counterparty_delayed_payment_base_key, counterparty_htlc_base_key, on_counterparty_tx_csv, per_htlc: HashMap::new() };
 
                let mut onchain_tx_handler = OnchainTxHandler::new(destination_script.clone(), keys.clone(), channel_static_info.clone());
```

Reply:
ah, that's probably leftovers, adopting your diff